modelon/impact/client/client.py

```diff
@@ -46,29 +47,36 @@ class Client:
     _SUPPORTED_VERSION_RANGE = ">=1.4.1,<2.0.0"
 
     def __init__(
-        self,
-        url=None,
-        interactive=None,
-        credentail_manager=None,
-        context=None,
+        self, url=None, interactive=None, credential_manager=None, context=None,
     ):
         if url is None:
             url = modelon.impact.client.configuration.get_client_url()
 
         if interactive is None:
             interactive = modelon.impact.client.configuration.get_client_interactive()
 
-        if credentail_manager is None:
-            credentail_manager = (
+        if credential_manager is None:
+            credential_manager = (
                 modelon.impact.client.credential_manager.CredentialManager()
             )
 
         self.uri = modelon.impact.client.sal.service.URI(url)
         self._sal = modelon.impact.client.sal.service.Service(self.uri, context)
-        self._credentials = credentail_manager
+        self._credentials = credential_manager
 
         self._validate_compatible_api_version()
-        self._authenticate_against_api(interactive)
+
+        try:
+            self._authenticate_against_api(interactive)
+        except modelon.impact.client.sal.exceptions.HTTPError:
+            if interactive:
+                logger.warning(
+                    "The provided API key is not valid, please enter a new key"
+                )
+                self.api_key = self._credentials.get_key_from_prompt()
```

Comment:
I think this don't work anymore because, `_authenticate_against_api` does not read from self.api_key anymore. Could probably add back so it reads from self.api_key and set self.api_key to None at the start of the constructor.

Reply:
The test goes through because it does not check that api/login is called twich with different API keys, only that get_key and get_key_from_prompt is called. Probably better if the test check what api/login is called with as that is what is important.